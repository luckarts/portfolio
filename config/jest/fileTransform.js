const path = require('path');

// This is a custom Jest transformer turning file imports into filenames.
// https://jest-bot.github.io/jest/docs/webpack.html

module.exports = {
  process(src, filename) {
    return `module.exports = ${JSON.stringify(path.basename(filename))};`;
  }
};