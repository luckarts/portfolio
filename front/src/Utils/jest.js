import checkPropTypes from "check-prop-types";
import { createStore } from "redux";
import rootReducer from "store/reducer";

export const checkProps = (component, expectedProps) => {
  // @ts-ignore
  const ErrProps = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return ErrProps;
};
export const testStore = initialState => {
  return createStore(rootReducer, initialState);
};
