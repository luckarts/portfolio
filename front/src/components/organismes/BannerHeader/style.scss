.banner-main {
	text-align: center;
	display: flex;
	flex-grow: 1;
	background-size: cover;
	background-image: url('/img/MainBg.jpg');
	height: 100vh;
	background-position: center;
}

.banner-main img {
	max-width: 90%;
	margin: auto;
}

.banner-inner {
	display: flex;
	width: 100%;
	background: linear-gradient(to top, #1b2735 0%, #090a0f 100%);
	//	background-color: rgba(15, 23, 31, 0.5);
	min-height: 100vh;
	overflow: hidden;

	.banner-container {
		display: flex;
		;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		margin: auto;
		animation: FadeSize .75s cubic-bezier(.17, .88, .32, 1.27) .75s both;


		h1 {
			margin-top: 5vh;
			font-size: 1.7rem;

		}


	}
}