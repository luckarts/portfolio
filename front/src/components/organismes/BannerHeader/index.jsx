import React, { useEffect } from 'react';

import PropTypes from 'prop-types';
import useWindowDimensions from 'Utils/useWindowDimensions';
import { Loader } from 'components/atoms';
import './style.scss';
import './stars.scss';
import { IoIosArrowRoundForward } from 'react-icons/io';
import { Button } from 'lucky-react-component';
import { Icon } from 'components/atoms/Icon';
const propTypes = {
	activeTab: PropTypes.func,
};

const BannerHeader = ({ activeTab }) => {
	const { width } = useWindowDimensions();

	useEffect(() => {
		const timer = setInterval(() => {
			if (width >= 1024) {
				activeTab(1);
			}
		}, 4000);

		return () => clearInterval(timer);
	}, [activeTab, width]);

	return (
		<div className="banner banner-main">
			<div className="banner-inner" onClick={() => activeTab(1)}>
				<div id="stars"></div>
				<div id="stars2"></div>
				<div id="stars3"></div>

				<div className="banner-container">
					<img src={'img/LOGO7.png'} alt="logo" />
					<h1>DÃ©veloppeur Fullstack Js</h1>
					{width >= 1024 ? (
						<Loader />
					) : (
						<Button color="primary" rounded="rounded" className="btn--arrow">
							Balayer{' '}
							<Icon>
								<IoIosArrowRoundForward />
							</Icon>
						</Button>
					)}
				</div>
			</div>
		</div>
	);
};

BannerHeader.propTypes = propTypes;

export default BannerHeader;
