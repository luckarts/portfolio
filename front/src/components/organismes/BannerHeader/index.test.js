import React from "react";
import { shallow, mount } from "enzyme";
import BannerHeader from "./index";
import { checkProps } from "Utils/jest";
let wrapper, props;
props = {
  activeTab: jest.fn()
};

it("Should render.", () => {
  expect(mount(<BannerHeader {...props} />)).toMatchSnapshot();
});

describe("BannerHeader Component", () => {
  beforeEach(() => {
    wrapper = shallow(<BannerHeader {...props} />);
    wrapper.find(".banner-inner").simulate("click");
  });

  it(" Checking PropTypes no error", () => {
    const PropsError = checkProps(BannerHeader, props);
    expect(PropsError).toBeUndefined();
  });
  it('Should call "openLightbox" when clicking on it', () => {
    expect(props.activeTab).toHaveBeenCalled();
  });
});
