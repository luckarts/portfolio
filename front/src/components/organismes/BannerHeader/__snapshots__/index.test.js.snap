// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Checking PropTypes renders correctly 1`] = `[Function]`;

exports[`Should render. 1`] = `ReactWrapper {}`;

exports[`renders correctly 1`] = `[Function]`;
